src/js/lib/youtube.js

```diff
@@ -6,52 +6,85 @@ import Tracker from './tracking';
 import {setStyles, scrollTo} from './dom';
 import {parse} from 'iso8601-duration';
 
-function pimpYouTubePlayer(videoId, node, height, width, chapters) {
-    const tracker = new Tracker({videoId: videoId});
-
-    const youtubePlayer = new YouTubePlayer(node.querySelector('#ytGuPlayer'), {
-        height: height,
-        width: width,
-        videoId: videoId,
-        playerVars: { 'autoplay': 0, 'controls': 1, 'rel': 0 }
-    });
+class PimpedYouTubePlayer {
+    play (seconds) {
+        const self = this;
+
+        self.tracker.track('play');
+
+        if (! isMobile()) {
+            self.el.querySelector('.docs__poster--wrapper').classList.add('docs__poster--wrapper--playing');
+        }
+
+        scrollTo(document.body, 0, 300);
+        self.el.querySelector('.docs__poster--loader').classList.add('docs__poster--hide');
+
+        if (seconds) {
+            self.player.seekTo(seconds, true);
+        }
+
+        self.player.playVideo();
+    }
 
-    youtubePlayer.on('ready', () => {
-        addChapterEventHandlers(node, youtubePlayer);
-        node.querySelector('.docs__poster--loader').addEventListener('click', function() {
-            playVideo(node.querySelector('.docs__poster--wrapper'), youtubePlayer, this);
+    pause () {
+        const self = this;
+
+        self.player.pauseVideo();
+    }
+
+    seekTo (seconds) {
+        const self = this;
+
+        self.play(seconds);
+    }
+
+    constructor(videoId, node, height, width, chapters) {
+        // declare `self` to avoid scoping issues of `this`
+        const self = this;
+
+        self.el = node;
+
+        self.tracker = new Tracker({videoId: videoId});
+
+        self.player = new YouTubePlayer(self.el.querySelector('#ytGuPlayer'), {
+            height: height,
+            width: width,
+            videoId: videoId,
+            playerVars: { 'autoplay': 0, 'controls': 1, 'rel': 0 }
         });
 
-        node.querySelector('.docs__shows-trailer').addEventListener('click', function() {
-            youtubePlayer.pauseVideo();
+        self.player.on('ready', () => {
+            addChapterEventHandlers();
+
+            self.el.querySelector('.docs__shows-trailer').addEventListener('click', () => self.pause());
+
+            self.player.getDuration().then(duration => self.videoDuration = duration);
         });
-    });
 
-    youtubePlayer.on('stateChange', event => {
-        let playTimer;
-        if (event.data === YT.PlayerState.PLAYING) {
-            youtubePlayer.getDuration().then(duration => {
```

Comment:
no point calculating the video duration on each tick, instead we [calculate it on `ready` and store it](https://github.com/guardian/docs-interactive-template/pull/80/files#diff-9e25e2d256d2580dd5abd4e38b2aef8eR61).